ghost/members-api/lib/repositories/member.js

```diff
@@ -726,9 +729,21 @@ module.exports = class MemberRepository {
         const memberIds = memberRows.map(row => row.id);
 
         if (data.action === 'unsubscribe') {
-            return await this._Member.bulkDestroy(memberIds, 'members_newsletters', {column: 'member_id'});
+            const hasNewsletterSelected = (Object.prototype.hasOwnProperty.call(data, 'newsletter') && data.newsletter !== null);
+            if (hasNewsletterSelected) {
+                const newsletters = await this._MemberNewsletter.where('newsletter_id', data.newsletter).fetchAll();
```

Comment:
Rather than use `reduce` below to get the correct list we should be able to make an SQL query to get the correct data.

So could do something like `this._MemberNewsletter.where().andWhereIn()` Or we can use the `getFilteredCollectionQuery` method above and pass in an NQL filter like `newsletter_id:blah+member_id:[id1,id2,id3...]`